### PR TITLE
Chore!: Migrate tenant datasource

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
**Note:** This change also includes updates to data source schemas generally moving things from optional -> read-only. These changes were mistakenly made in previous migrations. Changes are based on a comparison to previous docs -
https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/compare/release/0.22.0...isaac/tenant-datasource.

There are a number of properties that do not match their old schemas. These are generally the list of objects returned by the data source that have been defined as blocks. Blocks do not allow for Computed to be set. 

This is a list of what I've seen so far, it may not be extensive I just focused on data sources.
variables - Scope
Spaces - Spaces

__________________________________________________________________________________________________________________

```
data "octopusdeploy_tenants" "tenants" {
  skip = 0
  take = 10
}

output "octopusdeploy_tenants_out" {
  value = data.octopusdeploy_tenants.tenants
}
```

image

### CODE REVIEW

File: docs/data-sources/tenants.md [hunk 1]
```diff
@@ -18,31 +18,28 @@ Provides information about existing tenants.
 ### Optional
 
 - `cloned_from_tenant_id` (String) A filter to search for a cloned tenant by its ID.
+- `id` (String) The unique ID for this resource.
```
Comment (hnrkndrssn) [inline docs/data-sources/tenants.md:21]: This should be read-only as it's an ID set by us

File: docs/data-sources/tenants.md [hunk 2]
```diff
@@ -18,31 +18,28 @@ Provides information about existing tenants.
...
-### Read-Only
-
-- `id` (String) An auto-generated identifier that includes the timestamp when this data source was last modified.
 - `tenants` (Block List) A list of tenants that match the filter(s). (see [below for nested schema](#nestedblock--tenants))
 
 <a id="nestedblock--tenants"></a>
 ### Nested Schema for `tenants`
 
-Read-Only:
+Optional:
```
Comment (hnrkndrssn) [inline docs/data-sources/tenants.md:36]: These were all read-only previously...

**Review by hnrkndrssn — APPROVED** [review]
Changes looks good, just a few differences in the schema between read-only / optional attributes

File: docs/data-sources/tenants.md (hunk 2 above)
Comment (IsaacCalligeros95) [reply docs/data-sources/tenants.md:36]: Do you know why these change and how to keep them as read only?

File: docs/resources/azure_subscription_account.md [hunk 3]
```diff
@@ -22,9 +22,7 @@ resource "octopusdeploy_azure_subscription_account" "example" {
```
Comment (IsaacCalligeros95) [inline docs/resources/azure_subscription_account.md:22]: I've run the generate main.go for this PR and included all the latest docs changes. I can undo these resources changes if required, but at this stage I think it's easier to include the latest and if we have to compare to pre-migration schemas use  a [comparison against the release 0.22 branch](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/compare/release/0.22.0...isaac/tenant-datasource)

File: docs/data-sources/tenants.md (hunk 1 above)
Comment (IsaacCalligeros95) [reply docs/data-sources/tenants.md]: Resolved these issues and the same problem across most other data sources, except where blocks are used and I cannot set the Computed attribute.
image

**Review by hnrkndrssn — APPROVED** [review]

**Pull request merged**